sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -76,14 +93,42 @@ namespace Azure { namespace Core {
         std::string const& timeString,
         DateFormat format = DateFormat::Rfc1123);
 
+  private:
     /**
      * @brief Get a string representation of the @DateTime.
      *
      * @param format The representation format to use.
+     * @param fractionFormat The format for the fraction part of the Datetime. Only supported by
+     * ISO8601
      *
      * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
      */
-    std::string ToString(DateFormat format = DateFormat::Rfc1123) const;
+    std::string ToString(DateFormat format, TimeFractionFormat fractionFormat) const;
```

Comment:
I think it is strange to have this as another required option to `ToString()` because it doesn't scale well for more customization.

If we really need it, can we incorporate it into the DateTime format itself? .NET does it via DateTime format format strings.
https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings